### ISSUE TITLE
something in LUS is very expensive (probably a bug), reduces game to a standstill for me

### ISSUE THREAD

**Issue by silentwings** [SpringCabal/Gravitas#84]
See gadget profiler:
![screen00004](https://cloud.githubusercontent.com/assets/3885930/7247137/fb159a42-e7fd-11e4-8026-7bb2e10950e0.png)

Looks to me like the GameFrame event from (something in) LUS is killing us.

**Comment by Anarchid** [comment]
Collating stuff from thread.

> I've tried giving 100 gravit on a neutral map and having them jump, run and shoot around for 5 minutes with no notable loss of FPS. Needs more investigation.

---

> My first guess is that its related to the robots, based on how the profiler reacted to my game playing, but I haven't investigated much.

**Comment by silentwings** [comment]
We should check if lua mem increases indefinately, which, if it happens, may or not be the same issue.

**Comment by silentwings** [comment]
I isolated it via alot of hacking and testing. The cause is robots that are paralyzed (i.e. stuck in electrafi fields). If you want to check for yourself, open the electrafi gadget, turn off the paralysis damage to robots (line 137) and the gradual slowdown will not occur.

To reproduce easily, just using gravitas.sdd: 
(1) turn on all of gadget profiler (/luarules profile, /luarules uprofile, /luarules ap)
(2) spawn a couple of electrafi
(3) spawn 5 robots (i used bobs), walk them into the electrafi fields until they are paralyzed
(4) sit and wait, watch the synced LUS % slowly rise....
(5) If you look closely and turn on \debug you will also notice that on average lua mem usage slowly rises.

(4) & (5) are almost certainly caused by the same issue, and its very probably a bug in unitscript of the robots, probably triggered when and only when they become paralysed. 
If I had to take a long shot, without even reading the LUS (which I haven't) I would guess that aggressive/wasteful table/string creation/manipulation is the cause. It must be in something called via the LUS GameFrame() event, although I don't know how visible the call stack is to you when you are actually inside a script.

If you spawn some more electrafi and a sizeable number of robots, it will pretty quickly reduce your fps to a standstill.

P.S. I think they way robots shake before they become paralyzed in electrafi fields LOOKS awesome.
